routing/restriction_loader.cpp

```diff
@@ -15,13 +51,46 @@ RestrictionLoader::RestrictionLoader(MwmValue const & mwmValue)
     ReaderSource<FilesContainerR::TReader> src(*m_reader);
     m_header.Deserialize(src);
 
-    RestrictionSerializer::Deserialize(m_header, m_restrictions, src);
+    RestrictionVec restrictionsOnly;
+    RestrictionSerializer::Deserialize(m_header, m_restrictions /* restriction no */,
+                                       restrictionsOnly, src);
+    ConvertRestrictionsOnlyToNoAndSort(graph, restrictionsOnly, m_restrictions);
   }
   catch (Reader::OpenException const & e)
   {
     m_header.Reset();
     LOG(LERROR,
         ("File", m_countryFileName, "Error while reading", RESTRICTIONS_FILE_TAG, "section.", e.Msg()));
+    throw;
+  }
+}
+
+void ConvertRestrictionsOnlyToNoAndSort(IndexGraph const & graph,
+                                        RestrictionVec const & restrictionsOnly,
+                                        RestrictionVec & restrictionsNo)
+{
+  for (Restriction const & o : restrictionsOnly)
+  {
+    CHECK_EQUAL(o.m_featureIds.size(), 2, ("Only two link restrictions are support."));
```

Comment:
why not just skip? by doing skip the app can handle 3+ edges restriction that could appear in new data.